### ISSUE TITLE
Sir Shield (Platformer)

### ISSUE THREAD

**Issue by JamesOuO** [nathan/phosphorus#380]
Custom blocks (level generation + text) are much slower, anti-aliasing/smoothing is too much on magnified objects.

http://scratch.mit.edu/projects/55416430/
http://phosphorus.github.io/#55416430
Mozilla/5.0 (X11; CrOS armv7l 6310.68.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.96 Safari/537.36

**Comment by htmlgames** [comment]
This is the same issue as manic miner. If you add:

else source += 'VISUAL = false;\n'

after this blocks closing bracket, the rendering of the level speeds up considerably:

.. else if (['showBackground:', 'startScene', 'nextBackground', 'nextScene', 'startSceneAndWait', 'show', 'hide', 'putPenDown', 'stampCostume', 'showVariable:', 'hideVariable:', 'doAsk', 'setVolumeTo:', 'changeVolumeBy:', 'setTempoTo:', 'changeTempoBy:'].indexOf(block[0]) !== -1) {
        source += 'VISUAL = true;\n';
      }

Basically it hides the pen? scanning code, which swaps colored blocks for tiles on the collision layer.
However this can bring in other issues for other projects.

**Comment by htmlgames** [comment]
And regarding the canvas smoothing, please could you add in Stage.prototype.draw:

this.context.imageSmoothingEnabled = false;

(I noticed you had this fix in another branch)

**Comment by nathan** [comment]
> `else source += 'VISUAL = false;\n'`

That's just incorrect (i.e., not what Scratch does).

**Comment by htmlgames** [comment]
True, but because it demonstrates how I would have expected it to look for manic miner stage load (sadly it breaks other projects!) - I find it strange how a hide in the list is VISUAL = true, though I still working on decoding your phosphorus.js :). 

I'm wondering how Scratch implements this 'speedup' (i.e. some code quirk for this block layout order?).

Regarding the aliasing, will you be adding canvas imageSmoothingEnabled = false at all?

**Comment by htmlgames** [comment]
This is a run without screen refresh issue. Please see
https://github.com/nathan/phosphorus/issues/658
